src/shared/sleep-config.c

```diff
@@ -229,6 +245,50 @@ int find_hibernate_location(char **device, char **type, uint64_t *size, uint64_t
                                 continue;
                         }
 
+                        // Check if swap matches the resume= and resume_offset= values if configured.
+                        if (!streq(resume_offset, "0") && !streq(resume, "0:0")) {
+                                _cleanup_close_ int fd;
+                                _cleanup_free_ char *swap_device = NULL;
+                                struct stat sb;
+
+                                fd = open(swap->device, O_RDONLY|O_CLOEXEC|O_NOCTTY);
+                                if (!fd)
+                                        return log_error_errno(errno, "Failed to open %s: %m", swap->device);
+
+                                r = fstat(fd, &sb);
+                                if (r < 0)
+                                        return log_error_errno(errno, "Failed to stat %s: %m", swap->device);
+
+                                if (asprintf(&swap_device, "%u:%u", major(sb.st_dev), minor(sb.st_dev)) < 0)
+                                        return log_oom();
+
+                                r = btrfs_is_filesystem(fd);
+                                if (r < 0)
+                                        return log_error_errno(r, "Error checking %s for Btrfs filesystem: %m", swap->device);
+                                else if (r)
+                                        log_debug("Unable to detect offset on Btrfs: %s", swap->device);
+
+                                // If the resume device is the same as the swap file device, check the offset.
+                                if (streq(resume, swap_device)) {
```

Comment:
This should be an `else if` to the previous `if` clause to ensure this doesn't run if on `Btrfs`.